packages/protocol/src/dcl/DclCertificateService.ts

```diff
@@ -0,0 +1,583 @@
+/**
+ * @license
+ * Copyright 2022-2025 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    Bytes,
+    Construction,
+    Days,
+    Diagnostic,
+    Duration,
+    Environment,
+    Logger,
+    Seconds,
+    StorageContext,
+    StorageService,
+    Time,
+    Timer,
+} from "#general";
+import { Paa } from "../certificate/kinds/AttestationCertificates.js";
+import { DclClient, MatterDclError } from "./DclClient.js";
+import { DclPkiRootCertificateSubjectReference } from "./DclRestApiTypes.js";
+
+const logger = Logger.get("DclCertificateService");
+
+// GitHub repository for development/test certificates
+const GITHUB_OWNER = "project-chip";
```

Comment:
We have a github client in `support/codegen/src/util/github.ts`.  Maybe we could move that into `@matter/general` so it can be shared?